src/lib/gundb/gundb-extend.js

```diff
@@ -89,31 +89,24 @@ assign(User.prototype, {
    * @returns {Promise<any>}
    */
   async decrypt(callback = null) {
-    const user = this.back(-1).user()
-    const pair = user.pair()
-    let path = ''
-
-    this.back(({ is, get }) => {
-      if (is || !get) {
-        return
-      }
+    let decryptedData = null
+    const encryptedData = await this.then()
 
-      path += get
-    })
+    if (encryptedData) {
+      let path = ''
+      const user = this.back(-1).user()
+      const pair = user.pair()
 
-    const encryptedKey = await user
-      .get('trust')
-      .get(pair.pub)
-      .get(path)
-      .then()
+      this.back(({ is, get }) => !is && get && (path += get))
 
-    const secureKey = await SEA.decrypt(encryptedKey, pair)
-    const encryptedData = await this.then()
-    let decryptedData = null
+      const secureKey = await user
+        .get('trust')
+        .get(pair.pub)
+        .get(path)
+        .then(encryptedKey => (encryptedKey ? SEA.decrypt(encryptedKey, pair) : null))
 
-    if (encryptedData != null) {
       if (!secureKey) {
-        throw new Error(`Decrypting key missing for ${path} data:${encryptedData}`)
+        throw new Error(`Decrypting key missing for path '${path}'. Trying to decrypt '${encryptedData}'`)
```

Comment:
revert gundb-extend changes, it was fine as is

Reply:
1. we have to check if encryptedKey isn't empty before we call SEA.decrypt. Otherwise we'll get 'cb is not a function' error
2. logging encryptedData was added by you in some commit, it's on the master now. I kept this change in my PRs

i'll revert the other stuff

Reply:
i ment the other changes.
SEA now checks if data for decryption is null or not and  it simply  returns undefined. cb is not a function is a different error. dont throw error if data is undefined

Reply:
i dont see changes have been reverted

Reply:
1. if data is null it tries to call the callback should be passed as the third argument. it we don't pass anything (as we do here), this will produce uncaught "cb is not a function". so we have to a) do not decrypt if source data is null b) pass noop as callback 
2. @sirpy pls look changes from all commits.